server/executor/trigger/grpc.go

```diff
@@ -0,0 +1,197 @@
+package trigger
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/fullstorydev/grpcurl"
+	"github.com/golang/protobuf/jsonpb"
+	"github.com/golang/protobuf/proto"
+	"github.com/jhump/protoreflect/desc"
+	"github.com/kubeshop/tracetest/server/model"
+	"go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	"go.opentelemetry.io/otel/trace"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/status"
+)
+
+func GRPC() Triggerer {
+	return &grpcTriggerer{
+		traceProvider: traceProvider(),
+	}
+}
+
+type grpcTriggerer struct {
+	traceProvider *sdktrace.TracerProvider
+}
+
+func (te *grpcTriggerer) Trigger(_ context.Context, test model.Test, tid trace.TraceID, sid trace.SpanID) (Response, error) {
+
+	response := Response{
+		Result: model.TriggerResult{
+			Type: te.Type(),
+		},
+	}
+
+	trigger := test.ServiceUnderTest
+	if trigger.Type != model.TriggerTypeGRPC {
+		return response, fmt.Errorf(`trigger type "%s" not supported by GRPC triggerer`, trigger.Type)
+	}
+
+	if trigger.GRPC == nil {
+		return response, fmt.Errorf("no settings provided for GRPC triggerer")
+	}
+	var tf trace.TraceFlags
+	sc := trace.NewSpanContext(trace.SpanContextConfig{
+		TraceID:    tid,
+		SpanID:     sid,
+		TraceFlags: tf.WithSampled(true),
+		TraceState: trace.TraceState{},
+		Remote:     true,
+	})
+	ctx := trace.ContextWithSpanContext(context.Background(), sc)
+
+	tReq := trigger.GRPC
+
+	conn, err := dial(ctx, tReq.Address)
+	if err != nil {
+		return response, fmt.Errorf("cannot dial service: %w", err)
+	}
+
+	desc, err := protoDescription(tReq.ProtobufFile)
+	if err != nil {
+		return response, fmt.Errorf("cannot read descriptors: %w", err)
+	}
+
+	options := grpcurl.FormatOptions{
+		EmitJSONDefaultFields: true,
+		IncludeTextSeparator:  true,
+		AllowUnknownFields:    true,
+	}
+
+	rf, _, err := grpcurl.RequestParserAndFormatter(grpcurl.FormatJSON, desc, strings.NewReader(tReq.Request), options)
+	if err != nil {
+		return response, fmt.Errorf("failed to construct request parser and formatter for %q", err)
```

Comment:
I think you are either using the wrong variable or the wrong format option:

Either it's `%q` but the variable is not `err` or it should be `%w` and the variable is `error` indeed.

Reply:
good catch!